### PR TITLE
Add `ListTransactions` to Api with stub handler (returning [])

### PR DESCRIPTION
**Pull request by Anviking**
# Issue Number

#465 

# Overview

- [x] I have `ListTransactions` to the Api, and a stub handler which returns `[]`.
- [x] I have added a `"Range"` `Header` represented by a new `Iso8601Range "inserted-at"` type
- [x] I fixed the `HasPath` test machinery to work with `Header`s

# Comments

- `Iso8601Range` and its doc comments might be a bit crude, but I _think_ this is approximately how we want it.

### CODE REVIEW

**Comment by jonathanknowles** [comment]
@Anviking Thanks for adding this.

Would it be much extra work to also add the `Range` parameter? Even if this parameter is currently ignored, we'd still be able to immediately start connecting the CLI up to this endpoint.

> `string <inserted_at={range-start}-{range-end}>` 
> A range of date in ISO-8601 UTC format without symbols to retrieve.
> Note that the start joker symbol `*` can be used instead of a date to refer to a lower or upper boundary. > Example: `inserted-at 20190227T160329Z-*`

File: lib/core/src/Cardano/Wallet/Api.hs [hunk 1]
```diff
@@ -146,3 +158,33 @@ instance Accept Any where
...
+--    2019-02-27 and 2020-02-27, in ascending order.
+-- - `20200227T000000Z-20190227T000000Z`: means all transaction between
+--  2020-02-27 and 2019-02-27, in descending order.
+data Iso8601Range (name :: Symbol)
+    = Iso8601Range (Maybe UTCTime) (Maybe UTCTime)
+    deriving (Show)
+
+instance FromHttpApiData (Iso8601Range (name :: Symbol)) where
+    parseHeader = error "FromHttpApiData Iso8601Range to be implemented"
+    parseQueryParam = error "I guess we need to implement this too?"
```
Comment (KtorZ) [inline lib/core/src/Cardano/Wallet/Api.hs]: No we don't :)
Only one of the methods needs to be implemented, they're all implemented using one or the other. Note that this is the semantic behind "minimal complete definition" in the hackage documention:

https://hackage.haskell.org/package/http-api-data-0.4.1/docs/Web-HttpApiData.html#t:FromHttpApiData

Here it says:

<pre>
<b>Minimal complete definition</b>

<code>parseUrlPiece | parseQueryParam</code>
</pre>

Which means that you need to implement either `parseUrlPiece` or `parseQueryParam` and, other methods of the class are implicitly implemented :)

Comment (Anviking) [reply lib/core/src/Cardano/Wallet/Api.hs]: @KtorZ I looked at that! But here we want to implement **`parseHeader`**, which is **not** `parseUrlPiece` nor `parseQueryParam` and thus not a minimal complete implementation. And if you try anyway, it errors as expected. 😕

Comment (KtorZ) [reply lib/core/src/Cardano/Wallet/Api.hs]: Na, you can get away by just implementing `parseUrlPiece`. `parseHeader` in the end is just:

```hs
class FromHttpApiData a where
  {-# MINIMAL parseUrlPiece | parseQueryParam #-}
  -- | Parse URL path piece.
  parseUrlPiece :: Text -> Either Text a
  parseUrlPiece = parseQueryParam

  -- | Parse HTTP header value.
  parseHeader :: ByteString -> Either Text a
  parseHeader = parseUrlPiece <=< (left (T.pack . show) . decodeUtf8')

  -- | Parse query param value.
  parseQueryParam :: Text -> Either Text a
  parseQueryParam = parseUrlPiece
```

It's really just an alias. As long as you implement `parseUrlPiece`, you're fine.

Comment (Anviking) [reply lib/core/src/Cardano/Wallet/Api.hs]: Ah. Thanks.

File: lib/core/test/integration/Test/Integration/Scenario/API/Transactions.hs [hunk 2]
```diff
@@ -568,7 +568,7 @@ spec = do
         expectErrorMessage (errMsg404NoWallet $ w ^. walletId) r
 
     describe "TRANS_CREATE_08 - v2/wallets/{id}/transactions - Methods Not Allowed" $ do
-        let matrix = ["PUT", "DELETE", "CONNECT", "TRACE", "OPTIONS", "GET"]
+        let matrix = ["PUT", "DELETE", "CONNECT", "TRACE", "OPTIONS"]
```
Comment (Anviking) [inline lib/core/test/integration/Test/Integration/Scenario/API/Transactions.hs:571]: Had to remove this as we now have implemented `GET v2/wallets/{id}/transactions`.

Not sure if we should add a new test case for `ListTransactions` now or not.

**Comment by Anviking** [comment]
Might have to update my local jormungandr:integration setup (but relevant parts pass), and I think CI should pass now.

**Review by KtorZ — APPROVED** [review]

**Pull request merged**